src/loaders/lottie/tvgLottieModel.h

```diff
@@ -37,35 +37,47 @@ struct LottieStroke
 {
     struct DashAttr
     {
-        //0: offset, 1: dash, 2: gap
-        LottieFloat value[3] = {0.0f, 0.0f, 0.0f};
+        LottieFloat offset = 0.0f;
+        LottieFloat* value = nullptr;
+        uint8_t size = 0;
+        uint8_t allocated = 0;
     };
 
     virtual ~LottieStroke()
     {
+        if (dashattr) {
+            delete[] dashattr->value;
+            dashattr->value = nullptr;
+            dashattr->size = dashattr->allocated = 0;
+        }
         delete(dashattr);
     }
 
-    LottieFloat& dash(int no)
+
+    LottieFloat& dashValue()
     {
         if (!dashattr) dashattr = new DashAttr;
-        return dashattr->value[no];
-    }
 
-    float dashOffset(float frameNo, Tween& tween, LottieExpressions* exps)
-    {
-        return dash(0)(frameNo, tween, exps);
-    }
+        if (dashattr->size + 1 > dashattr->allocated) {
+            dashattr->allocated = dashattr->size + 2;
 
-    float dashSize(float frameNo, Tween& tween, LottieExpressions* exps)
-    {
-        auto d = dash(1)(frameNo, tween, exps);
-        return (d > 0.0f) ? d : 0.0f;
+            LottieFloat* newValue = new LottieFloat[dashattr->allocated];
+            for (uint8_t i = 0; i < dashattr->size; ++i) {
+                newValue[i].copy(dashattr->value[i], false);
```

Comment:
Doesn't this work with a swallow copy?
`newValue[i] = LottieFloat(dashattr->value[i]);`